webapp/src/ts/services/simprints.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class SimprintsService {
+  currentRequest:any = {};
+  SP_ID_MASK = 0xFFFFF8;
+  MAX_TIER = 3;
+  resolvePromise;
+  rejectPromise;
+
+  private request(endpoint) {
+    const requestId = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER) & this.SP_ID_MASK;
+    this.currentRequest = {
+      id: requestId,
+      deferred: new Promise((resolve, reject) => {
+        this.resolvePromise = resolve;
+        this.rejectPromise = reject;
```

Comment:
Looks like you never call `rejectPromise`. Is it needed?

Reply:
Not needed. I have removed it.